app/controllers/admin/api_users_controller.rb

```diff
@@ -0,0 +1,68 @@
+class Admin::ApiUsersController < ApplicationController
+  before_action :find_api_user, only: %i[show generate_token destroy]
+
+  def index
+    @api_users = ApiUser.order(created_at: :asc).page(params[:page])
+  end
+
+  def new
+    @api_user = ApiUser.new
+  end
+
+  def create
+    @api_user = ApiUser.new(api_user_params.merge(created_by: current_user.name))
+
+    if @api_user.save
+      redirect_to admin_api_user_path(@api_user), notice: notice('successfully.created')
+    else
+      flash.now.alert = notice('could_not_be.created')
+      render :new
+    end
+  end
+
+  def show; end
+
+  def generate_token
+    timestamp = Time.current
+    token = generate_api_user_token(timestamp)
+    if token && @api_user.update({ token_sample: token.last(5),
+                                   token_generated_at: timestamp,
+                                   token_generated_by: current_user.name })
```

Comment:
why just not user id and than read user name ?

Reply:
Primarily this is in case the user that generated the token is subsequently deleted.

Reply:
makes sense